SeaBattle/BaseShip.cs

```diff
@@ -0,0 +1,148 @@
+                Decks.Add(a);
+            }
+        }
+        public override string ToString()
+        {
+            StringBuilder output = new StringBuilder ("Name: ", 25);
+            output.Append (Name + ", ");
+            output.Append("Type: " + this.GetType() + ", ");
+            output.Append("Durability: " + Durability + ", ");
+            output.Append("Speed: " + Speed + ", ");
+            output.Append("Decks:" + Decks.Count + ", ");
+            output.Append("Coordinates: ");
+            foreach (var deck in Decks)
+            {
+                output.Append(deck.Location.X + ":" + deck.Location.Y + ", ");
+            }
+            return output.ToString();
+        }
+        public static bool operator== (BaseShip s1, BaseShip s2)
+        {
+            return ((s1.Decks.Count.Equals(s2.Decks.Count)) && (s1.GetType().Equals(s2.GetType())));
+        }
+        public static bool operator!= (BaseShip s1, BaseShip s2)
+        {  
+            return !((s1.Decks.Count.Equals(s2.Decks.Count)) && (s1.GetType().Equals(s2.GetType())));
+
+        }
+        public bool Equals (BaseShip s1)
+        {
+            return ((this.Decks.Count == s1.Decks.Count) && (this.GetType() == s1.GetType()));
+
+        }
+
+        // This function adds s1(BaseShip) to the Sea
+        // 1. + Checking if Point within Sea borders
+        // 2. + Checking if Point is available 
+        public void AddShip(BaseShip s1, Sea PlaceSea)
+        {
+            foreach (var d in s1.Decks)
+            {
+                if (d.Location.X > PlaceSea.SeaWidth || d.Location.Y > PlaceSea.SeaHeight || d.Location.X < 0 || d.Location.Y < 0)
+
+                    throw new Exception("Ship can not exsist with this Sea borders");
+            }
+            if (ShipSpaceCheck(s1.Decks, PlaceSea))
+            {
+                PlaceSea.Ships.Add(s1);
+            }
+        }
+
+        // To do in the MoveTo method  
+        // 1. + calculating mid-ship coordinates of current position
+        // 2. + calculating mid-ship coordinates of new position
+        // 3. + calcualating distance of the move
+        // 4. + checking if speed is enough to get to new position
+        // 5. + checking if new position is available
+        // 6. + updating current position with new position
+        public void MoveTo(List<Deck> NewPosition, Sea Sea1)
+        {
+            float midShipX = 0;
+            float midShipY = 0;
+            foreach (var d in Decks)
+            {
+                midShipX += d.Location.X;
+                midShipY += d.Location.Y;
+            }
+            midShipX = midShipX / Decks.Count;
+            midShipY = midShipY / Decks.Count;
+
+            float midNewX = 0;
+            float midNewY = 0;
+            foreach (var p in NewPosition)
+            {
+                midNewX += p.Location.X;
+                midNewY += p.Location.Y;
+            }
+            midNewX = midNewX / NewPosition.Count;

```

Comment:
you can use compound assignment here too, as you did above with the += operator

Reply:
Fixed.